File: Code/0070.Climbing Stairs/Climbing Stairs_1.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [1, 2]
        if n <= 2:
            return dp[n-1]
        for i in range(3, n+1):
            dp[0], dp[1] = dp[1], dp[0]+dp[1]       # 相较于下面代码而言，节约了空间成本，不需要储存整个爬楼梯的情况
            # dp[i] = dp[i-1] + dp[i-2]
        return dp[1]

File: Code/0070.爬楼梯/Climbing Stairs_1.java
class Solution {
    public int climbStairs(int n) {
        // 使用动态规划的方法做，与LC509题类似，斐波那契数列
        if(n <= 2){
            return n;
        }

        int[] dp = new int[2];
        // 初始化动态数组
        dp[0] = 1;
        dp[1] = 2;
        // 动态传递
        int temp;
        for(int i = 2; i < n; i++){
            temp = dp[1];
            dp[1] = dp[0] + dp[1];
            dp[0] = temp;

            // dp[i] = dp[i-1] + dp[i-2];
        }
        return dp[1];
    }
}